docker-compose.yml

```diff
@@ -412,8 +412,12 @@ services:
     volumes:
       - ./scripts/security:/etc/kafka/secrets
     environment:
-      KSQL_CONFIG_DIR: "/etc/ksql"
-      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties"
+      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j.properties"
```

Comment:
Needs to be `file:/etc/ksql-server/log4j.properties`

Reply:
@mikebin good catch, and wow this means all the other examples are wrong too!

```
examples(5.4.x): grep -ir KSQL_LOG4J_OPTS * | grep file:\/    
ccloud/clients/ksql-datagen/docker-compose.yml:      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties"
ccloud/clients/ksql-datagen/docker-compose.yml:      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties"
ccloud/docker-compose.yml:      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties"
clients/cloud/ksql-datagen/docker-compose.yml:      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties"
clients/cloud/ksql-datagen/docker-compose.yml:      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties"
cp-all-in-one/docker-compose.yml:      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties"
cp-all-in-one/docker-compose.yml:      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties"
cp-all-in-one-cloud/docker-compose.yml:      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties"
cp-all-in-one-community/docker-compose.yml:      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties"
cp-all-in-one-community/docker-compose.yml:      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties"
microservices-orders/docker-compose.yml:      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties"
multi-datacenter/docker-compose.yml:      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-silent.properties"
multi-datacenter/docker-compose.yml:      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-silent.properties"
multi-datacenter/docker-compose.yml:      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-silent.properties"
music/docker-compose.yml:      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties"
postgres-debezium-ksql-elasticsearch/docker-compose/docker-compose.yml:      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties"
postgres-debezium-ksql-elasticsearch/docker-compose/docker-compose.yml:      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties"
```

Reply:
@mikebin I don't understand log4j well enough to understand why we would want one over the other.  Looking at the diff, can you elaborate?

```
$ docker-compose exec ksql-server diff /etc/ksql/log4j.properties /etc/ksql-server/log4j.properties
1,15d0
< #
< # Copyright 2018 Confluent Inc.
< #
< # Licensed under the Confluent Community License (the "License"); you may not use
< # this file except in compliance with the License.  You may obtain a copy of the
< # License at
< #
< # http://www.confluent.io/confluent-community-license
< #
< # Unless required by applicable law or agreed to in writing, software
< # distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
< # WARRANTIES OF ANY KIND, either express or implied.  See the License for the
< # specific language governing permissions and limitations under the License.
< #
< 
20,24c5
< log4j.appender.stdout.layout.ConversionPattern=[%d] %p %m (%c:%L)%n
< 
< log4j.appender.streams=org.apache.log4j.ConsoleAppender
< log4j.appender.streams.layout=org.apache.log4j.PatternLayout
< log4j.appender.streams.layout.ConversionPattern=[%d] %p %m (%c:%L)%n
---
> log4j.appender.stdout.layout.ConversionPattern=[%d] %p %m (%c)%n
26,32d6
< log4j.logger.kafka=ERROR, stdout
< log4j.logger.org.apache.kafka.streams=INFO, streams
< log4j.additivity.org.apache.kafka.streams=false
< log4j.logger.org.apache.zookeeper=ERROR, stdout
< log4j.logger.org.apache.kafka=ERROR, stdout
< log4j.logger.org.I0Itec.zkclient=ERROR, stdout
< log4j.logger.org.reflections=ERROR, stdout
34,35d7
< # Uncomment the following line to stop KSQL from logging out each request it receives:
< #log4j.logger.io.confluent.ksql.rest.server.resources.KsqlResource=WARN, stdout
```

Reply:
Previously, the main reason to pick `/etc/ksql-server/log4j.properties` over `/etc/ksql/log4j.properties` was because environment variable substitution was done on `/etc/ksql-server/log4j.properties`: https://github.com/confluentinc/ksql-images/blob/5.3.1-post/cp-ksql-server/include/etc/confluent/docker/configure#L32.

However, if we're now going to provide a full properties file as a volume mount to include security settings, instead of relying on environment variable substitution, it doesn't really matter which location we choose, as long as we configure accordingly in `KSQL_LOG4J_OPTS`

Reply:
And actually, I think KSQL will by default look for `/etc/ksql/log4j.properties`, so that is probably the preferred location, to avoid having to set `KSQL_LOG4J_OPTS`. But if we're going to provide a custom file as a volume mount, we might need `KSQL_LOG4J_OPTS` anyway, if the file won't be named `/etc/ksql/log4j.properties`.

Reply:
Thanks for the explanation @mikebin .  In the latest version of this PR, the file is renamed in order to emphasize that it is not using the default (it nudges users to explore it if they are examining the KSQL server configuration).